Files/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -598,19 +598,22 @@ public async Task<ReturnResult> CopyItemsFromClipboard(DataPackageView packageVi
                 return ReturnResult.Failed;
             }
 
-            if (source.Any())
+            if (!source.IsEmpty())
             {
                 ReturnResult returnStatus = ReturnResult.InProgress;
 
-                var destinations = new List<string>();
-                List<ShellFileItem> binItems = null;
+                List<string> destinations = new List<string>();
                 foreach (var item in source)
                 {
                     if (recycleBinHelpers.IsPathUnderRecycleBin(item.Path))
                     {
-                        binItems ??= await recycleBinHelpers.EnumerateRecycleBin();
-                        var matchingItem = binItems.FirstOrDefault(x => x.RecyclePath == item.Path); // Get original file name
-                        destinations.Add(PathNormalization.Combine(destination, matchingItem?.FileName ?? item.Name));
+                        List<ShellFileItem> binItems = await recycleBinHelpers.EnumerateRecycleBin();
```

Comment:
I agree with the null check below but EnumerateRecycleBin really shouldn't be put inside the loop.